src/Runner.Worker/JobExtension.cs

```diff
@@ -74,6 +74,10 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                         {
                             // print out HostName for self-hosted runner
                             context.Output($"Runner name: '{setting.AgentName}'");
+                            if (message.Variables.TryGetValue("system.runnerGroupName", out VariableValue runnerGroupName))
```

Comment:
do you need to check `!string.IsNullOrEmpty(runnerGroupName.Value)`

Reply:
We outright don't set this variable on the payload so I think this should be sufficient. @lokesh755 is going to test so should be able to say